src/org/zaproxy/zap/extension/proxies/OptionsProxiesPanel.java

```diff
@@ -57,10 +60,19 @@ public OptionsProxiesPanel(ExtensionProxies extension) {
 
     private void initialize() {
         this.setName(Constant.messages.getString("proxies.options.title"));
+        this.add(getScrollPane());
         this.setLayout(new GridBagLayout());
         this.add(getMainProxyPanel(), LayoutHelper.getGBC(0, 0, 1, 0.0));
         this.add(getSecurityProtocolsPanel(), LayoutHelper.getGBC(0, 1, 1, 0.0));
-        this.add(getProxiesOptionsPanel(), LayoutHelper.getGBC(0, 2, 1, 1.0, 1.0));
+        this.add(getScrollPane(), LayoutHelper.getGBC(0, 2, 1, 1.0, 1.0));
+    }
+    
+    private JScrollPane getScrollPane() {
+        if (proxiesScrollPane == null) {
+            proxiesScrollPane = new JScrollPane();
+            proxiesScrollPane.setViewportView(getProxiesOptionsPanel());

```

Comment:
Although it requires some more changes I'd suggest adding to the whole panel instead of just the table (as it would be always scrollable no matter the size of the dialogue).